app/src/features/toolbar/components/SearchBar.tsx

```diff
@@ -24,19 +24,34 @@ function SearchBar() {
   const theme = useTheme();
   const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
   const [searchValue, setSearchValue] = useState(
-    () => searchParams.get("query") || "",
+    () => searchParams.get("q") || "",
   );
   const debounceTimeoutRef = useRef<NodeJS.Timeout>();
 
   const updateURL = useCallback(
     (value: string) => {
       const trimmed = value.trim();
-      const url = trimmed
-        ? `/?query=${encodeURIComponent(trimmed)}&page=1`
-        : "/";
+      const newParams = new URLSearchParams();
+      
+      // Preserve existing category and keyword filters
+      const category = searchParams.get("category");
```

Comment:
nit: i see these `"q"`, `"category"` and `"keyword"` reused quite often across the UI; could be worthwhile making these consts and then imported